src/components/key_pad_menu/key_pad_menu_item.tsx

```diff
@@ -269,14 +276,17 @@ export const EuiKeyPadMenuItem: FunctionComponent<EuiKeyPadMenuItemProps> = ({
   }
 
   return (
-    <Element
-      className={classes}
-      {...(relObj as ElementType)}
-      {...(rest as ElementType)}
-      // Unable to get past `LegacyRef` conflicts
-      ref={buttonRef as Ref<any>}
-    >
-      {renderContent()}
-    </Element>
+    <div className="euiKeyPadMenuItem__outer">
+      <Element
+        className={classes}
+        {...(relObj as ElementType)}
+        {...(rest as ElementType)}
+        // Unable to get past `LegacyRef` conflicts
+        ref={buttonRef as Ref<any>}
+      >
+        {renderContent()}
+      </Element>
+      {betaBadgeLabel && !checkable && renderBetaBadge()}
```

Comment:
The `betaBadgeLabel` check is already in `renderBetaBadge()` so I'd just move the `!checkable` check into there as well
```suggestion
      {renderBetaBadge()}
```